### PR TITLE
split flux, add docs

### PR DESCRIPTION
**Pull request by charleskawczynski**
- Split the `AtmosModel` flux, into components that match the design docs.
 - Added documentation

### CODE REVIEW

**Comment by charleskawczynski** [comment]
bors try

File: src/Atmos/Model/AtmosModel.jl [hunk 1]
```diff
@@ -52,25 +52,59 @@ function vars_aux(m::AtmosModel, T)
...
+```
+∂Y
+-- = - ∇ • F + S(Y)
+∂t
+```
+Where
+```
+F =   F_{adv}      Advective flux                                  , see flux_advective!    for this term
+    + F_{press}    Pressure flux                                   , see flux_pressure!     for this term
+    + F_{nonidff}  Fluxes that do *not* contain gradients          , see flux_nondiffusive! for this term
```
Comment (tapios) [inline src/Atmos/Model/AtmosModel.jl]: Typo: `nondiff`

Comment (tapios) [inline src/Atmos/Model/AtmosModel.jl]: Typo `nondiff`

**Comment by charleskawczynski** [comment]
bors try

**Comment by charleskawczynski** [comment]
bors try

**Comment by charleskawczynski** [comment]
@simonbyrne should we allow codecov to drop? I think Julia is optimizing away the empty function calls.

**Pull request merged**